crates/bevy_render/src/view/visibility/mod.rs

```diff
@@ -26,31 +27,42 @@ use crate::{
 
 /// If an entity is hidden in this way,  all [`Children`] (and all of their children and so on) will also be hidden.
 /// This is done by setting the values of their [`ComputedVisibility`] component.
-#[derive(Component, Clone, Reflect, Debug)]
+#[derive(Component, Clone, Reflect, Debug, PartialEq, Eq)]
 #[reflect(Component, Default)]
-pub struct Visibility {
-    /// Indicates whether this entity is visible. Hidden values will propagate down the entity hierarchy.
-    /// If this entity is hidden, all of its descendants will be hidden as well. See [`Children`] and [`Parent`] for
-    /// hierarchy info.
-    pub is_visible: bool,
+pub enum Visibility {
+    Shown,
+    Hidden,
 }
 
 impl Default for Visibility {
     fn default() -> Self {
-        Visibility::VISIBLE
+        Self::Shown
     }
 }
 
-impl Visibility {
-    /// A [`Visibility`], set as visible.
-    pub const VISIBLE: Self = Visibility { is_visible: true };
+impl Not for &Visibility {
+    type Output = Visibility;
+
+    #[inline]
+    fn not(self) -> Visibility {
+        match self {
+            Visibility::Shown => Visibility::Hidden,
+            Visibility::Hidden => Visibility::Shown,
+        }
+    }
+}
 
-    /// A [`Visibility`], set as invisible.
-    pub const INVISIBLE: Self = Visibility { is_visible: false };
+impl Visibility {
+    /// Whether this entity is visible.
+    #[inline]
+    pub const fn is_visible(&self) -> bool {
+        matches!(self, Self::Shown)
+    }
 
     /// Toggle the visibility.
+    #[inline]
     pub fn toggle(&mut self) {
-        self.is_visible = !self.is_visible;
+        *self = !&*self
```

Comment:
@mockersf this `*self = !&*self` is the best I can do I think without adding `Copy`

Reply:
I would totally add `Copy` to this struct :)